recipe/meta.yaml

```diff
@@ -50,6 +55,10 @@ build:
     - echo $LLAMA_ARGS  # [unix]
     - set LLAMA_ARGS    # [win]
 
+    - export CUDACXX=/usr/local/cuda/bin/nvcc   # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
+    - export CUDAHOSTCXX="${CXX}"               # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
+    - export LDFLAGS="$LDFLAGS -Wl,-rpath-link,/usr/lib64"    # [(cuda_compiler_version or "").startswith("12")]
```

Comment:
Will we have to do this even every feedstock that use cuda 12?

Reply:
Setting CUDACXX and CUDAHOSTCXX doesn't apply here because cuda_compiler_version=12.3 and selector has not startswith 12.
I believe the LDFLAGS is set to link to a cuda library not provided in the cuda packages. To be confirmed by Daniel.

Reply:
Was it because of cuda_driver? https://github.com/ggerganov/llama.cpp/blob/b2408/CMakeLists.txt#L392
If so, a comment in the recipe would be welcome.

Reply:
yes, for libcuda.so, the cuda driver. will add a comment.
Not sure about the requirement for other feedstocks. Can have a closer look if necessary. Probably a good idea. Can we move that out of scope of this PR since it's internal delivery only? I can look at it in the scope of the rest of the CUDA 12 work

Reply:
>Can we move that out of scope of this PR since it's internal delivery only? I can look at it in the scope of the rest of the CUDA 12 work

Yep, makes sense to me. My comment was more a general question anyway and I was not expecting a fix now. I was curious to know if we missed something or if this is required because our environment is different than conda-forge or something else.

So I would say, let's keep an eye on this and if we have to do this again in another feedstock, then we could take some time to investigate why it's needed?

Reply:
https://anaconda.atlassian.net/browse/PKG-4306